controllers/vspherecluster_reconciler.go

```diff
@@ -359,7 +359,7 @@ func (r *clusterReconciler) reconcileDeploymentZones(ctx context.Context, cluste
 }
 
 func (r *clusterReconciler) reconcileClusterModules(ctx context.Context, clusterCtx *capvcontext.ClusterContext) (reconcile.Result, error) {
-	if feature.Gates.Enabled(feature.NodeAntiAffinity) {
+	if feature.Gates.Enabled(feature.NodeAntiAffinity) && !clusterCtx.VSphereCluster.Spec.DisableClusterModule {
```

Comment:
What about the other places where the feature gate is used?

Reply:
@sbueringer,
Added the missing condition in the vspherevm_controller.go.
Didn't add the missing condition in `AddClusterControllerToManager` function in the vspherecluster_controller.go as this is a one time config and can not be changed per cluster

Reply:
Makes sense, thx!